patches/server/1052-add-number-format-api.patch

```diff
@@ -0,0 +1,175 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: David Mayr <davidliebtkekse@gmail.com>
+Date: Sat, 16 Dec 2023 10:40:29 +0100
+Subject: [PATCH] add number format api
+
+
+diff --git a/src/main/java/io/papermc/paper/util/PaperScoreboardFormat.java b/src/main/java/io/papermc/paper/util/PaperScoreboardFormat.java
+new file mode 100644
+index 0000000000000000000000000000000000000000..a9e2d3fe4420bad2bbd2d37d04411fa1c2ad74d9
+--- /dev/null
++++ b/src/main/java/io/papermc/paper/util/PaperScoreboardFormat.java
+@@ -0,0 +1,55 @@
++package io.papermc.paper.util;
++
++
++import io.papermc.paper.scoreboard.numbers.NumberFormat;
++
++public class PaperScoreboardFormat {
++
++    private PaperScoreboardFormat() {}
++
++
++    public static net.minecraft.network.chat.numbers.NumberFormat asVanilla(NumberFormat format) {
++        net.minecraft.network.chat.numbers.NumberFormat vanilla;
++        if (format instanceof io.papermc.paper.scoreboard.numbers.StyledFormat styled) {
++            com.google.gson.JsonElement json = net.kyori.adventure.text.serializer.gson.GsonComponentSerializer.gson().serializer()
++                .toJsonTree(styled.style());
++            net.minecraft.network.chat.Style vanillaStyle = net.minecraft.Util
++                .getOrThrow(net.minecraft.network.chat.Style.Serializer.CODEC
++                    .parse(com.mojang.serialization.JsonOps.INSTANCE, json), com.google.gson.JsonParseException::new);
++
++            vanilla = new net.minecraft.network.chat.numbers.StyledFormat(vanillaStyle);
++        } else if (format instanceof io.papermc.paper.scoreboard.numbers.FixedFormat fixed) {
++            vanilla = new net.minecraft.network.chat.numbers.FixedFormat(io.papermc.paper.adventure.PaperAdventure
++                .asVanilla(fixed.component()));
++        } else if(format instanceof io.papermc.paper.scoreboard.numbers.BlankFormat) {
++            vanilla = net.minecraft.network.chat.numbers.BlankFormat.INSTANCE;
++        } else {
++            throw new IllegalArgumentException("Unknown format type " + format.getClass());
++        }
++
++        return vanilla;
++    }
++
++    public static NumberFormat asPaper(net.minecraft.network.chat.numbers.NumberFormat vanilla) {
++        if (vanilla instanceof net.minecraft.network.chat.numbers.StyledFormat styled) {
++            com.google.gson.JsonElement json = net.minecraft.Util
++                .getOrThrow(
++                    net.minecraft.network.chat.Style.Serializer.CODEC
++                        .encodeStart(com.mojang.serialization.JsonOps.INSTANCE, styled.getStyle()),
++                    com.google.gson.JsonParseException::new
++                );
++
++            net.kyori.adventure.text.format.Style adventureStyle =
++                net.kyori.adventure.text.serializer.gson.GsonComponentSerializer.gson()
++                    .serializer().fromJson(json, net.kyori.adventure.text.format.Style.class);
```

Comment:
As `jpenilla` said, use the style codec from AdventureCodec to convert between the two.